artistools/inputmodel/__init__.py

```diff
@@ -173,9 +173,36 @@ def vectormatch(vec1, vec2):
         abundancedata = get_initialabundances(modelpath)
         dfmodeldata = dfmodeldata.merge(abundancedata, how='inner', on='inputcellid')
 
+    if derived_cols:
+        add_derived_cols_to_modeldata(dfmodeldata, derived_cols, dimensions, t_model_init_seconds, wid_init, modelpath)
+
     return dfmodeldata, t_model_init_days, vmax_cmps
 
 
+def add_derived_cols_to_modeldata(dfmodeldata, derived_cols, dimensions=None, t_model_init_seconds=None, wid_init=None,
+                                  modelpath=None):
+    """add columns to modeldata using e.g. derived_cols = ('velocity', 'Ye')"""
+    if dimensions == 3 and 'velocity' in derived_cols:
+        dfmodeldata['vel_x'] = dfmodeldata['pos_x'] / t_model_init_seconds
+        dfmodeldata['vel_y'] = dfmodeldata['pos_y'] / t_model_init_seconds
+        dfmodeldata['vel_z'] = dfmodeldata['pos_z'] / t_model_init_seconds
+
+        dfmodeldata['vel_x_outer'] = (dfmodeldata['pos_x'] + wid_init) / t_model_init_seconds
+        dfmodeldata['vel_y_outer'] = (dfmodeldata['pos_y'] + wid_init) / t_model_init_seconds
+        dfmodeldata['vel_z_outer'] = (dfmodeldata['pos_z'] + wid_init) / t_model_init_seconds
```

Comment:
I think this would only be the outer (i.e. further from 0,0,0) x/y/z velocity for positive x/y/z coords. For negative x, vel_x_outer would actually be the inner velocity (because it's on the positive x direction side of the cube). If it was me that named 'pos_x' then I regret it. A better choice is the min/mid/max coordinates/velocities.